cmake/GazeboUtils.cmake

```diff
@@ -181,7 +181,7 @@ macro (gz_setup_windows)
     endif()
 
     if (MSVC)
-      add_compile_options(/Zc:__cplusplus)
+      add_compile_options(/Zc:__cplusplus /permissive- /Zc:strictStrings-)
```

Comment:
the windows CI build is broken; I'm guessing it's related to this change?

* [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gazebo-ci-pr_any-windows7-amd64&build=3405)](https://build.osrfoundation.org/job/gazebo-ci-pr_any-windows7-amd64/3405/) https://build.osrfoundation.org/job/gazebo-ci-pr_any-windows7-amd64/3405/

~~~
[ 43%] Building CXX object gazebo/transport/CMakeFiles/gazebo_transport.dir/IOManager.cc.obj
C:\Jenkins\workspace\gazebo-ci-pr_any-windows7-amd64\ws\install\include\boost/interprocess/detail/win32_api.hpp(892): error C2116: 'GetProcessTimes': function parameter lists do not match between declarations
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\processthreadsapi.h(106): note: see declaration of 'GetProcessTimes'
C:\Jenkins\workspace\gazebo-ci-pr_any-windows7-amd64\ws\install\include\boost/interprocess/detail/win32_api.hpp(892): error C2733: 'GetProcessTimes': you cannot overload a function with 'extern "C"' linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\processthreadsapi.h(106): note: see declaration of 'GetProcessTimes'
C:\Jenkins\workspace\gazebo-ci-pr_any-windows7-amd64\ws\install\include\boost/interprocess/detail/win32_api.hpp(908): error C2371: 'GetFileType': redefinition; different basic types
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\fileapi.h(708): note: see declaration of 'GetFileType'
C:\Jenkins\workspace\gazebo-ci-pr_any-windows7-amd64\ws\install\include\boost/interprocess/detail/win32_api.hpp(908): error C2733: 'GetFileType': you cannot overload a function with 'extern "C"' linkage
...
~~~

Reply:
I pushed a change. Hopefully that should fix it.

Reply:
There are test failures, but those seem to have been happening for a while. Both tests seem to fail because they are using command not available on Windows cmd (`make` and `rm`). Would it be worth modifying them or removing them?